src/NCronJob/Execution/StartupJobManager.cs

```diff
@@ -8,12 +11,45 @@ internal class StartupJobManager(
     public async Task ProcessStartupJobs(CancellationToken stopToken)
     {
         var startupJobs = jobRegistry.GetAllOneTimeJobs();
-        var startupTasks = startupJobs.Select(definition => CreateExecutionTask(JobRun.Create(observer.Report, definition), stopToken)).ToList();
 
-        if (startupTasks.Count > 0)
+        if (startupJobs.Count == 0)
         {
-            await Task.WhenAll(startupTasks).ConfigureAwait(false);
+            return;
         }
+
+        List<JobRun> jobRuns = [];
+        var startupTasks = startupJobs.Select(definition =>
+        {
+            var jobRun = JobRun.Create(observer.Report, definition);
+            jobRuns.Add(jobRun);
+            return CreateExecutionTask(jobRun, stopToken);
+        });
+
+        await Task.WhenAll(startupTasks).ConfigureAwait(false);
+
+        Exception?[] faults = jobRuns
+            .Where(jr => jr.JobDefinition.ShouldCrashOnStartupFailure == true && jr.CurrentState.Type == JobStateType.Faulted)
+            .Select(jr => jr.CurrentState.Fault)
+            .ToArray();
+
+        if (faults.Length == 0)
+        {
+            return;
+        }
+
+        StringBuilder sb = new();
+        sb.Append("At least one of the startup jobs failed");
```

Comment:
Nitpicky: But using `AppendLine here, would remove the `Environment.NewLine` stuff on line 49

Reply:
Fixing previous comment would also cover this, I believe.

Reply:
Fixed